Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/storage/retention/TraceLocationRetention.kt

```diff
@@ -0,0 +1,21 @@
+package de.rki.coronawarnapp.eventregistration.storage.retention
+
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.TraceLocation
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.seconds
+import de.rki.coronawarnapp.util.TimeStamper
+import java.util.concurrent.TimeUnit
+
+private const val TRACE_LOCATION_RETENTION_DAYS = 15
+private val TRACE_LOCATION_RETENTION_SECONDS = TimeUnit.DAYS.toSeconds(TRACE_LOCATION_RETENTION_DAYS.toLong())
+
+fun isWithinRetention(traceLocation: TraceLocation, timeStamper: TimeStamper): Boolean {
```

Comment:
1- `TraceLocation. isWithinRetention` ?
2- Pass the instant so it can be common between all TraceLocations?
3- some unit tests would be great ?

Reply:
Fixed all three points.